regular supervisor wrapping dynamic supervisor always propagate restarts

stuartish opened the pull request:
connect #215

cc: @derekkraan

derekkraan commented:
Hmm, I think we're only halfway there with this PR. What we want is to be able to specify in `child_spec/1` of `Horde.DynamicSupervisor` the max_restarts and max_seconds, such that when this limit is reached for children added with `Horde.DynamicSupervisor.add_child/3`, the `Horde.DynamicSupervisor` dies.

With this change, the max_restarts and max_seconds is always 5/3. Thoughts?

stuartish commented:
@derekkraan I added max_restarts and max_seconds to the options for the ProcessesSupervisor.

Also:
- changed some names so we weren't going from max_seconds to intensity and then back to max_seconds 🤷
- removed duplicate `strategy` from options passed to DynamicSupervisorImpl
- added tests
